Package Manager Local Dependencies Draft

aciidgh opened the pull request.

abertelrud commented on proposals/xxxx-package-manager-local-dependencies.md:
```diff
@@ -0,0 +1,72 @@
+
+## Motivation
+
+Edit mode allows developing on set of packages without first committing and
```
Rather than focus on edit mode here, I would suggest motivating this by observing that in some cases the relationship between two packages is most accurately described using a local path.  This could be because the packages are still in "bring-up" mode, because they are so closely linked that it doesn't make sense to version one without the other, etc.  The edit mode can be mentioned but I wouldn't present this as a band-aid on the transient nature of edit mode, but rather that in some cases, the local path mode is simply the right conceptual relationship.

abertelrud commented on proposals/xxxx-package-manager-local-dependencies.md:
```diff
@@ -0,0 +1,72 @@
+its sources will be used as-is. There are no restrictions on the path and it can
+be relative or absolute.
+
+All local dependencies should be in the root package or reachable via other
```
I think one obvious question that will come up for people is the case of wanting multiple packages within the same repo, in which case they might want to have the top-level package in the repo have a local reference to a "wholly owned" package that is at some subpath inside the repo.  Since we're not intending to start the multi-package repo discussion here, we should make explicit that this is the reason we're disallowing local path references inside remote package manifests, and that a more comprehensive proposal for multi-package repos will follow in the future.

aciidb0mb3r commented:
@abertelrud Thanks, updated.

johnno1962 committed:
Raw String escaping

johnno1962 committed:
Raw string literals

jrose-apple committed:
Non-exhaustive enums: Rename to '@frozen' (and "non-frozen")

Credit to Brent Royal-Gordon for this name!

jrose-apple committed:
Non-exhaustive enums: add `unknown case`

Per review feedback.

jrose-apple committed:
Non-exhaustive enums: address feedback, mostly on `unknown case`

- Mention that this applies to both Swift and C enums
- Explain why this is only relevant across module boundaries
- Add an example of using `fallthrough` with `unknown case`
- Add Scala under "Languages with alternate designs"
- Describe what happens if a library author breaks the `@frozen` contract
- Add `case #unknown` as an alternative spelling
- Add the "only applies to ABI-stable libraries" alternative
- Add a stub for a "leave out `unknown case`" alternative
- Expand on what would happen if you mixed `unknown case` and `default`
- Add the "new declaration kind" alternative
- Add the "use protocols instead" alternative
- Add the "C enums should be RawRepresentable structs" alternative
- Fix typos

(Not all of this feedback is new to the `unknown case` revision.)

jrose-apple committed:
Non-exhaustive enums: Contrast a custom case with a custom pattern

It's weird to have a custom case, but it's even weirder to have a
custom pattern that you can only use in one place. As Chris pointed
out on the list, the main reason for the restriction on the pattern's
use is that the proposal makes it a /warning/ to leave known cases
unhandled, not an error. Therefore, leave the proposal describing
`unknown case` (or `unknown default`), but mention that we'd probably
switch to `case #unknown` if we were going to treat this as an error.

jrose-apple committed:
Non-exhaustive enums: spend more time talking about `static import`

(which would treat all enums in a particular module as frozen)

Per Dave DeLong's feedback, though I'm not sure this will satisfy him
either. :-) Still, might as well write it down.

jrose-apple committed:
Non-exhaustive enums: Add "Differences from the first revision"

Just to make second-round review easier.

harlanhaskins committed:
Update syntax with parentheses

harlanhaskins committed:
Add previous revision link

cwakamo committed:
[Proposal] Replace PlaygroundQuickLook/CustomPlaygroundQuickLookable with new API (#784)

Kicking off review for SE-0198: Playground Quicklook Revamp

tkremenek committed:
Merge pull request #785 from harlanhaskins/diagnostic-directives

[SE-0196] Update syntax with parentheses.

tkremenek committed:
Accept 196, and provide rationale.

rjmccall committed:
Accept SE-0197 with revisions.

DougGregor committed:
SE-0143 "Conditional conformances" is implemented in Swift 5.

tkremenek committed:
Accept SE-0195

airspeedswift committed:
[common rejections] Add ! to the list of logic operators, clarify C heritage (#788)

chriseidhof committed:
Bool toggle proposal (#782)

* Bool toggle proposal

* Suggestions by @xwu

* Feature name

* Update and rename nnnn-bool-toggle.md to 0199-bool-toggle.md

krilnon committed:
Add implementation PR to SE-0199.

krilnon committed:
[Status] Constrain detail items to keep columns aligned

aciidgh committed:
Add local dependencies draft

jeis2497052 committed:
Prpose fix a typo

krilnon committed:
Merge pull request #792 from jeis2497052/master

Fix a typo.

airspeedswift committed:
Update 0198-playground-quicklook-api-revamp.md

airspeedswift committed:
Add “with revisions”

Link to revision diff TK

xwu committed:
[gardening] Update SE-0195 and SE-0196 for implementation status (#794)

* Update SE-0195 for implementation status

* Update SE-0196 for implementation status

* Update index.html for copyright date

* Update index.css for copyright date

* Update index.js for copyright date

xwu committed:
[gardening] Correct the Swift version for which SE-0195 and SE-0196 are implemented (#795)

* SE-0195 is implemented in Swift 4.2

* SE-0196 is implemented in Swift 4.2

krilnon committed:
[Status] Add 4.2 to version filter.

jtbandes committed:
Revise SE-0194

jtbandes committed:
remove exclusion of deprecated cases

airspeedswift committed:
Update for accepted review

tkremenek committed:
Extend review of SE-0193

linqingmo committed:
Correct the Swift version for SE-0143 (#800)

RenanGreca committed:
Update swift-3_0.md

Fixed a typo (soure -> source).

krilnon committed:
Merge pull request #802 from RenanGreca/patch-1

Fix typo in swift-3_0.md

jrose-apple committed:
Non-exhaustive enums: Cut down SE-0192 per core team feedback

The biggest difference is that it will no longer apply to Swift enums
defined outside the stdlib and overlays; as such, it no longer
provides a spelling for declaring frozen enums. The feature previously
known as `unknown case` is now just spelled `unknown` and can match
any pattern containing a non-frozen enum, or even a pattern containing
an "implicitly frozen enum" from a third-party library (to help with
gradual adoption of changes within a project). See the list of changes
from the first proposal for more information.

tkremenek committed:
Merge pull request #777 from jrose-apple/non-exhaustive-enums-2

SE-0192 Non-exhaustive enums: add `unknown` and rename to '@frozen'

tkremenek committed:
Extend review of SE-0192 for a second phase.

tkremenek committed:
Fix review date format for 192

DougGregor committed:
SE-0194 "Derived Collection of Enum Cases" is implemented for Swift 4.2

DougGregor committed:
Initiate review of SE-0200: "Raw" mode string literals

airspeedswift committed:
Update 0199-bool-toggle.md

norio-nomura committed:
Fix typo (#805)

benrimmington committed:
[SE-0200] Remove pipermail links

stephencelis committed:
Discourse updates (#803)

* Discourse updates

* Update README.md

benrimmington committed:
[0000-template.md] Use forums instead of pipermail

krilnon committed:
Fix the status format in SE-0199 to match the template.

airspeedswift committed:
Update 0198-playground-quicklook-api-revamp.md

neonichu committed:
Update SE number
